src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,434 @@
+            await util.executeAndThrowIfError(`${dockerTool} ${command}`)
+
+            // Read the temp file to get the runtime stack into a variable
+            let oryxRuntimeTxtPath = path.join(appSourcePath, 'oryx-runtime.txt');
+
+            let runtimeStack = fs.promises.readFile(oryxRuntimeTxtPath, 'utf8').then((data) => {
+                let lines = data.split('\n');
+                return lines[0];
+            }).catch((err) => {
+                toolHelper.writeError(err.message);
+                throw err;
+            });
+
+            // Delete the temp file
+            fs.unlink(oryxRuntimeTxtPath, (err) => {
+                if (err) {
+                    toolHelper.writeWarning(`Unable to delete the temporary file "${oryxRuntimeTxtPath}". Error: ${err.message}`);
+                }
+            });
+
+            return runtimeStack;
+        } catch (err) {
+            toolHelper.writeError(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Sets the default builder on the machine to the Oryx++ Builder to prevent an exception from being thrown due
+     * to no default builder set.
+     */
+    public async setDefaultBuilder() {
+        toolHelper.writeInfo('Setting the Oryx++ Builder as the default builder via the pack CLI');
+        try {
+            let command = `config default-builder ${ORYX_BUILDER_IMAGE}`
+            await util.executeAndThrowIfError(`${PACK_CMD} ${command}`);
+        }
+        catch (err) {
+            toolHelper.writeError(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Installs the pack CLI that will be used to build a runnable application image.
+     * For more Information about the pack CLI can be found here: https://buildpacks.io/docs/tools/pack/
+     */
+    public async installPackCliAsync() {
+        toolHelper.writeDebug('Attempting to install the pack CLI');
+        try {
+            let command: string = '';
+            let commandLine = '';
+            if (IS_WINDOWS_AGENT) {
+                let packZipDownloadUri: string = 'https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-windows.zip';
+                let packZipDownloadFilePath: string = path.join(PACK_CMD, 'pack-windows.zip');
+                command = `New-Item -ItemType Directory -Path ${PACK_CMD} -Force | Out-Null; Invoke-WebRequest -Uri ${packZipDownloadUri} -OutFile ${packZipDownloadFilePath}; Expand-Archive -LiteralPath ${packZipDownloadFilePath} -DestinationPath ${PACK_CMD}; Remove-Item -Path ${packZipDownloadFilePath}`;
+                commandLine = 'pwsh';
+            } else {
+                let tgzSuffix = os.platform() == 'darwin' ? 'macos' : 'linux';
+                command = `(curl -sSL \"https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-${tgzSuffix}.tgz\" | ` +
```

Comment:
nit: same comment here about escaping the double quotes -- I know this one is escape in the Azure DevOps Task, but I'm pretty sure it doesn't need to be since we're using tick marks to wrap the string (I might have been using double quotes to wrap the string previously and forgot to change)

Reply:
Removed double quotes.